tst/standard/SubfieldMatrixGroups.tst

```diff
@@ -71,3 +71,25 @@ gap> testsOrthogonalSubfieldSU := [[0, 3, 5], [0, 5, 3], [-1, 2, 3], [-1, 4, 3]]
 #@fi
 gap> ForAll(testsOrthogonalSubfieldSU, TestOrthogonalSubfieldSU);
 true
+gap> TestSubfieldSp := function(args)
+>   local n, p, e, f, G, hasSize;
+>   n := args[1];
+>   p := args[2];
+>   e := args[3];
+>   f := args[4];
+>   G := SubfieldSp(n, p, e, f);
+>   hasSize := HasSize(G);
+>   RECOG.TestGroup(G, false, Size(G));
+>   return IsSubset(Sp(n, p ^ e), GeneratorsOfGroup(G))
+>          and DefaultFieldOfMatrixGroup(G) = GF(p ^ e)
+>          and hasSize;
+> end;;
+gap> testsSubfieldSp := [[6, 2, 2, 1], [4, 3, 2, 1], [4, 3, 4, 2], [4, 7, 2, 1]];;
+gap> ForAll(testsSubfieldSp, TestSubfieldSp);
+true
+gap> SubfieldSp(3, 2, 2, 1);
```

Comment:
I would suggest to add a comment that explains error handling is being tested (as opposed to a broken test file)
```suggestion

# test error handling
gap> SubfieldSp(3, 2, 2, 1);
```